### ISSUE TITLE
How to validate AJAX with CSRF ? 

### ISSUE THREAD

**Issue by Istanbouly** [slimphp/Slim-Csrf#72]
Hello guys I have a little problem I am using slim 3 framework along with slim-csrf what I did is registering the csrf to the container I aslo created a csrfExtension that extends the twigExtension.
in the csrfExtenstion class I created a method that return the csrf fields neede markup this way I don't need to retype the whole markup in my forms I just call the function that I created and it works fine.

Now the problem is in my website admins-panel page there is a lot of ajax requests that I want to apply the csrf on here is where the problem appears the csrf only works for the first request but it will give an error on any other request cause the csrf_name and csrf_value that I am picking from my view has expired so how to get the new csrf_name and csrf_value so I can pass them along with my other ajax requests.

hope you can help me
Thanks in advance

**Comment by jeffmthomas** [comment]
Not sure if it was the proper way to do it but my dirty hack was to create a GET route that I could call with ajax and retrieve CSRF tokens, then inject those CSRF tokens into your post request before you send it.

**Comment by Istanbouly** [comment]
Yeah Correct I thought if this but its not the proper way I did the same but I think that if someone know that route I will face the same problem cause the this get route will not be protected with CSRF right

**Comment by jeffmthomas** [comment]
But those generated CSRF tokens are still only valid for one use and have an expiration, which can be obtained anyways by simply refreshing any page that uses CSRF.

**Comment by jeffmthomas** [comment]
Another option would be to enable the persistent tokens.

**Comment by Istanbouly** [comment]
Sorry for late reply.
What are presistent Tokens

**Comment by jeffmthomas** [comment]
I believe it is the 6th option when you instantiate slim-csrf. 

_Sent from my Google Pixel XL using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by mridah** [comment]
@edwardteach42  can you show us the initialization code ? 

I currently have the default code:

    $container['csrf'] = function ($c) {
        return new \Slim\Csrf\Guard;
    };
